addons/mail/static/src/models/composer_view.js

```diff
@@ -505,10 +505,14 @@ registerModel({
          * @param {ClipboardEvent} ev
          */
         async onPasteTextInput(ev) {
-            if (!ev.clipboardData || !ev.clipboardData.files) {
-                return;
+            if (ev.clipboardData) {
+                if (ev.clipboardData.getData && ev.clipboardData.getData('text')) {
+                    this.composer.refreshLinkPreview(ev.clipboardData.getData('text'));
+                }
+                if (ev.clipboardData.files.length > 0) {
+                    await this.fileUploader.uploadFiles(ev.clipboardData.files);
+                }
```

Comment:
I haven't tested in practice, but at a glance of code, it dynamically updated link preview just on pasted text.
That means all other input interaction will preserve the preview.

The problem is that adapting link previews is costly right now, so doing it for all user interactions would introduce performance issues.

I think we should throttle refresh of link preview somehow, otherwise it's strange to still see outdated link previews